### DISCUSSION TITLE
Configuring OpenSuse Leap 15.5 with WSL

### DISCUSSION THREAD

**Discussion by FRacine** [microsoft/WSL#10562] [Q&A]
Hi,

We are installing Open Suse Leap 15.5 on Windows 10.

At first start of OpenSuse leaf, we are asking for an account creation. Is it possible bypassing that windows with a commandline then creating the account from a windows commandline?

It is possible closing the window then on next start we are on root but the account were not get created :(

Thanks

**Comment by sbradnick** [comment]
The user account creation is part of 'firstboot'. There is a "Skip User Creation" radio button, but I can't tell if that's what you're referring to with "Is it possible bypassing that ..."

If you're saying you did bypass the firstboot uid creation during the initial install:
- then set a root password during firstboot,
- create a new user via powershell through wsl (something like: `PS> wsl useradd -u 1000 -g users -m -d /home/someUser -l someUser`)
- you'd then (or also) have to put something like:
```
[user]
default=someUser
```
in `/etc/wsl.conf`.

Some good info here: https://learn.microsoft.com/en-us/windows/wsl/wsl-config

**Comment by FRacine** [comment]
On the first boot, it is the administrator user to be create 
So if I close the Window, I will have to create the administor account and a user with a commandline late.
Is it possible with a commandline to not have the initial windows at all the  configuring it later  without the need of just killing the window?

**Comment by sbradnick** [comment]
I don't follow what you're trying to accomplish, or really what you're trying to avoid and accomplish in some other way. The firstboot setup is there for convenience, to follow generally agreed upon standards for working in the openSUSE ecosystem; well, the Linux general convention is not use root directly, but a non-privileged user with sudo abilities, which firstboot sets up.

You can install openSUSE Leap 15.5 (or possibly use `wsl --install openSUSE-Leap-15.5 --web-download --no-launch`, but I'm not sure how doing that, then `wsl useradd ...` might work since it's not technically installed) and either create a "sudo" user which will be used when the distro boots, or you can set a root password (and use it when the distro boots). You can also use `wsl useradd ...` as I outlined above to create whatever basic user you want (provided you use `useradd` w/ the options that would result in the desired outcome).

I can't tell if you're saying you do an install and at the "Welcome" screen just close the window; that seems like a similar action to "Abort". So you'd be dropped to a `root@<hostname>` prompt when you launch the distro w/o having a root password (so you'd want to set it anyway).

Maybe some more good info here: https://learn.microsoft.com/en-us/windows/wsl/basic-commands

**Reply by FRacine** [reply]
Do you have a reference about wsl useradd as I don't find it.

**Comment by sbradnick** [comment]
I'm not suggesting that `wsl useradd` is a command, I'm saying that from PowerShell you can invoke wsl to directly run the (Linux) useradd command using the flags related to it:
```
PS> wsl useradd -u 1000 -g users -m -d /home/someUser -l someUser
```

That would be the case if you've exited out of the DistroLauncher.exe for openSUSE-Leap-15.5 as you've been describing doing ("So if I close the Window ..."). You could easily just be sitting at a root shell and run `useradd -u 1000 -g users -m -d /home/someUser -l someUser` instead.
